library/std/src/thread/mod.rs

```diff
@@ -185,29 +185,27 @@ mod scoped;
 #[stable(feature = "scoped_threads", since = "1.63.0")]
 pub use scoped::{scope, Scope, ScopedJoinHandle};
 
+mod current;
+
+#[stable(feature = "rust1", since = "1.0.0")]
+pub use current::current;
+pub(crate) use current::{drop_current, set_current, try_current};
+
 ////////////////////////////////////////////////////////////////////////////////
 // Thread-local storage
 ////////////////////////////////////////////////////////////////////////////////
 
 #[macro_use]
 mod local;
 
-cfg_if::cfg_if! {
-    if #[cfg(test)] {
-        // Avoid duplicating the global state associated with thread-locals between this crate and
-        // realstd. Miri relies on this.
```

Comment:
With https://github.com/rust-lang/miri/pull/3739 merged, this is no longer true.

Reply:
I sure hope so.^^ But it might still be worth doing this change separately just to be sure. The relevant test only runs at bors time.

Reply:
I've split this out into #128135.